PluralKit.Bot/Commands/SystemEdit.cs

```diff
@@ -523,9 +533,44 @@ public async Task Delete(Context ctx, PKSystem target)
             throw new PKError(
                 $"System deletion cancelled. Note that you must reply with your system ID (`{target.Hid}`) *verbatim*.");
 
-        await ctx.Repository.DeleteSystem(target.Id);
+        // If the user confirms the deletion, export their system and send them the export file before actually
+        // deleting their system, unless they specifically tell us not to do an export.
+
+        var noExport = ctx.MatchFlag("ne", "no-export");
+        if (!noExport)
+        {
+            var json = await ctx.BusyIndicator(async () =>
+            {
+                // Make the actual data file
+                var data = await _dataFiles.ExportSystem(ctx.System);
+                return JsonConvert.SerializeObject(data, Formatting.None);
+            });
+
+            var stream = new MemoryStream(Encoding.UTF8.GetBytes(json));
+
+            try
+            {
+                var dm = await _dmCache.GetOrCreateDmChannel(ctx.Author.Id);
+                var msg = await ctx.Rest.CreateMessage(dm,
+                    new MessageRequest { Content = $"{Emojis.Success} System deleted. If you want to set up your PluralKit system again, you can import the file below with `pk;import`." },
+                    new[] { new MultipartFile("system.json", stream, null) });
+                await ctx.Rest.CreateMessage(dm, new MessageRequest { Content = $"<{msg.Attachments[0].Url}>" });
+
+                // If the original message wasn't posted in DMs, send a public reminder
+                if (ctx.Channel.Type != Channel.ChannelType.Dm)
+                    await ctx.Reply($"{Emojis.Success} System deleted. Check your DMs for a copy of your system's data!");
+            }
+            catch (ForbiddenException)
+            {
+                // If user has DMs closed, tell 'em to open them
+                throw new PKError(
```

Comment:
Can you reword this error? If someone with DMs closed is running `pk;s delete`, don't know about the export, and get this error, it doesn't make sense. Maybe have it tell them it tried to send an export file when deleting, and to use the flag if they just want to do it anyway.